orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/CaptureSourceServerGroupCapacityTask.groovy

```diff
@@ -40,28 +40,35 @@ class CaptureSourceServerGroupCapacityTask implements Task {
     def stageOutputs = [:]
     StageData stageData = stage.mapTo(StageData)
     if (stageData.useSourceCapacity) {
-      def sourceServerGroup = oortHelper.getTargetServerGroup(
-        stageData.source.account,
-        stageData.source.asgName,
-        stageData.source.region,
-        stageData.cloudProvider ?: stageData.providerType
-      ).orElse(null)
-
-      if (sourceServerGroup) {
-        // capture the source server group's capacity AND specify an explicit capacity to use when deploying the next
-        // server group (ie. no longer use source capacity)
+      if (!stageData.source && stageData.preferSourceCapacity) {
```

Comment:
`stageData.source` gets set by the upstream `DetermineSourceServerGroupTask` (or it's null if it can't be found).

Not sure if that makes sense or not. I've run this locally as a pipeline in a few permutations and everything seems to work as expected.